src/test/kotlin/it/pagopa/wallet/scheduler/services/SchedulerLockServiceTest.kt

```diff
@@ -0,0 +1,114 @@
+package it.pagopa.wallet.scheduler.service
+
+import it.pagopa.wallet.scheduler.config.properties.RedisJobLockPolicyConfig
+import it.pagopa.wallet.scheduler.exceptions.LockNotAcquiredException
+import it.pagopa.wallet.scheduler.exceptions.LockNotReleasedException
+import it.pagopa.wallet.scheduler.exceptions.SemNotAcquiredException
+import it.pagopa.wallet.scheduler.exceptions.SemNotReleasedException
+import kotlin.test.Test
+import org.junit.jupiter.api.BeforeAll
+import org.mockito.ArgumentMatchers
+import org.mockito.kotlin.*
+import org.redisson.Redisson
+import org.redisson.api.RLockReactive
+import org.redisson.api.RPermitExpirableSemaphoreReactive
+import org.redisson.api.RedissonReactiveClient
+import org.redisson.config.Config
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import reactor.core.publisher.Mono
+import reactor.kotlin.test.test
+import redis.embedded.RedisServer
+
+class SchedulerLockServiceTest {
+    private val logger: Logger = LoggerFactory.getLogger(javaClass)
+    private val rLockReactive: RLockReactive = mock()
+    private val rPermitExpirableSemaphoreReactive: RPermitExpirableSemaphoreReactive = mock()
+
+    @Test
+    fun `Should acquire lock and release after work`() {
+        val jobName = "job-name-lock"
```

Comment:
add verify methods to check expected methods to be called once with wanted parameters

Reply:
do this for all the below tests please :D